### DISCUSSION TITLE
How to deal with nested lists where base list item should be a specific type?

### DISCUSSION THREAD

**Discussion by GlenNicholls** [pydantic/pydantic#4372] [Question] [answered]
I am trying to create a model for a YAML config file and one thing I would like to support is anchors. However, I have no idea how to define a nested list type in Pydantic. Take the following YAML for example

``` yaml
a1: &a1
- item1
- item2

a2: &a2
- *a1
- item4

stuff:
- *a1
- *a2
- itemX
```

So how do I create a model for `stuff` where it is a list where the base type of each item should be a string. In the above example, I wouldn't always know how deeply nested `stuff` items are. With JSON schema, I'm able to define a recursive type that states the type is itself or string, but Pydantic doesn't seem to like that when I do something like

``` py
class NestedList(BaseModel):
  __root__ : List[Union[str, NestedList]]

class ConfigSchema(BaseModel):
  stuff: NestedList
```

That gives me an error about `NestedList` being defined. However, if I add quotes:

``` py
class NestedList(BaseModel):
  __root__ : List[Union[str, "NestedList"]]
```

It's isn't validating properly as it's not throwing an error when I add an integer, bool, or dict to the list.

**Comment by samuelcolvin** [comment, marked as answer]
Should work fine.

https://pydantic-docs.helpmanual.io/usage/postponed_annotations/
